packages/postgres-database/src/simcore_postgres_database/utils_projects_nodes.py

```diff
@@ -16,24 +17,24 @@
 #
 # Errors
 #
-class BaseProjectNodesError(Exception):
-    ...
+class BaseProjectNodesError(PydanticErrorMixin, RuntimeError):
+    msg_template: str = "Project nodes unexpected error"
 
 
-class ProjectNodesProjectNotFound(BaseProjectNodesError):
-    ...
+class ProjectNodesProjectNotFoundError(BaseProjectNodesError):
+    msg_template: str = "Project {project_uuid} not found"
 
 
-class ProjectNodesNodeNotFound(BaseProjectNodesError):
-    ...
+class ProjectNodesNodeNotFoundError(BaseProjectNodesError):
+    msg_template: str = "Node {node_id!r} from project {project_uuid!r} not found"
 
 
-class ProjectNodesOperationNotAllowed(BaseProjectNodesError):
-    ...
+class ProjectNodesNonUniqueNodeFoundError(BaseProjectNodesError):
+    msg_template: str = "Multiple project found containing node {node_id}. TIP: misuse, the same node ID was found in several projects."
 
 
-class ProjectNodesDuplicateNode(BaseProjectNodesError):
-    ...
+class ProjectNodesDuplicateNodeError(BaseProjectNodesError):
+    msg_template: str = "Project node already exists, you cannot have 2x the same node in the same project."
```

Comment:
I think this reads a bit better
```suggestion
    msg_template: str = "Project node already exists, you cannot have 2x the same node_id in the same project."
```